pkg/operator/operator.go

```diff
@@ -122,6 +122,13 @@ func NewOperator(ctx context.Context, operator *operator.Operator) (context.Cont
 	} else {
 		logging.FromContext(ctx).With("cluster-endpoint", clusterEndpoint).Debugf("discovered cluster endpoint")
 	}
+	if clusterCIDR, err := ResolveClusterCIDR(ctx, eks.New(sess)); err != nil {
```

Comment:
Could go either way here, this will be tested implicitly by the AL2023 end-to-end tests since a CIDR is required for the node to connect to the cluster. If we have prior art for testing field discovery like this explicitly (e.g. Cluster Endpoint) we may as well add it as well but otherwise I think the implicit test is sufficient.

Reply:
We do have unit test and prior art with doing that for clusterEndpoint